orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/DeregisterInstancesFromLoadBalancerTaskSpec.groovy

```diff
@@ -42,40 +43,42 @@ class DeregisterInstancesFromLoadBalancerTaskSpec extends Specification {
 
   def "creates a deregister instances from load balancer task based on job parameters"() {
     given:
-      def operations
-      task.kato = Mock(KatoService) {
-        1 * requestOperations("abc", _) >> {
-          operations = it[1]
-          rx.Observable.from(taskId)
-        }
+    def operations
+    task.trafficGuard = Mock(TrafficGuard)
+    task.kato = Mock(KatoService) {
+      1 * requestOperations("abc", _) >> {
+        operations = it[1]
+        rx.Observable.from(taskId)
       }
+    }
 
     when:
     task.execute(stage)
 
     then:
-      operations.size() == 1
-      with(operations[0].deregisterInstancesFromLoadBalancer) {
-        it instanceof Map
-        instanceIds == this.deregisterInstancesFromLoadBalancerConfig.instanceIds
-        loadBalancerNames == this.deregisterInstancesFromLoadBalancerConfig.loadBalancerNames
-        region == this.deregisterInstancesFromLoadBalancerConfig.region
-        credentials == this.deregisterInstancesFromLoadBalancerConfig.credentials
-        cloudProvider == this.deregisterInstancesFromLoadBalancerConfig.cloudProvider
-      }
+    operations.size() == 1
+    with(operations[0].deregisterInstancesFromLoadBalancer) {
```

Comment:
worth doing an assertion that the mocked `TrafficGuard` was at least called with the parameters you expected?

Reply:
thanks, added in a6b7ebb83e4fa9df2f0337812de9906777a22973